### PR TITLE
Chitter Challenge Submission

### PR DESCRIPTION
**Pull request by RBGeomaticsRob**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+require 'bcrypt'
+class User
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  include DataMapper::Resource
+
+  validates_confirmation_of :password
```
Comment (RBGeomaticsRob) [inline app/models/user.rb]: Not able to get this to work as hound requests, I think it may be Rails specific I'll keep looking for a Datamapper example

File: spec/peep_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+require 'spec_helper'
+
+describe Peep do
+  it 'it expects to be created and retrieved from the database' do
+    user = User.create(name: 'Rob',
```
Comment (RBGeomaticsRob) [inline spec/peep_spec.rb]: Still investigating how to effectively stub this out, the datamapper save_parents method is causing me a problem as I cannot find a way to instigate the relationship using a double, I can respond to the command, but not get the code doing what is should, which is sort of making me think that if I achieved this I would be testing the complexity of the stub rather than the code in peep?

File: .rubocop.yml [hunk 3]
```diff
@@ -67,4 +76,13 @@ Style/Semicolon:
...
+  AllowAsExpressionSeparator: false
+
+Metrics/LineLength:
+  Max: 120
+  # To make it possible to copy or click on URIs in the code, we allow lines
+  # contaning a URI to be longer than Max.
+  AllowURI: true
+  URISchemes:
+    - http
+    - https
```
Comment (sjmog) [inline .rubocop.yml]: Nice hamstringing :)

File: README.md [hunk 4]
```diff
@@ -32,37 +18,44 @@ So that I can avoid others posting messages on Chitter as me
...
+The app runs on a Postgres database, you will need to create two databases `chitter_test` & `chitter_development`.
 
-* In order to start a conversation as a maker I want to reply to a peep from another maker.
+You will also need to `bundle install` to obtain all the necessary gems.
 
-And/Or:
+##Contribution##
 
-* Work on the css to make it look good (we all like beautiful things).
+This is a challenge set to identify progress in Wk3 at Makers Academy, therefore is designed to be tackled individually.
```
Comment (sjmog) [inline README.md]: Wk5!

File: Rakefile [hunk 5]
```diff
@@ -1,9 +1,22 @@
...
 RuboCop::RakeTask.new :cop
 RSpec::Core::RakeTask.new :spec
 Cucumber::Rake::Task.new :cuke
 
-task default: [:cop, :spec, :cuke]
\ No newline at end of file
+task default: [:cop, :spec, :cuke]
+
+task :auto_upgrade do
+  require './app/data_mapper_setup'
```
Comment (sjmog) [inline Rakefile]: I'd really like to find a better way to do this - to avoid using `require` within a task

File: app/chitter.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
+require 'sinatra/base'
+require 'data_mapper'
+require 'rack-flash'
+require 'sinatra/partial'
+require_relative 'data_mapper_setup'
```
Comment (sjmog) [inline app/chitter.rb]: Why is this here?

File: app/chitter.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+class Chitter < Sinatra::Base
+
+  register Sinatra::Partial
+  set :partial_template_engine, :erb
```
Comment (sjmog) [inline app/chitter.rb]: V excited to see your usage of Partials later :)

File: app/chitter.rb [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+  helpers do
+    def maker
+      @maker ||= User.get(session[:user_id]) if session[:user_id]
+    end
+  end
+
+  require './app/controllers/index'
+  require './app/controllers/chits'
+  require './app/controllers/users'
```
Comment (sjmog) [inline app/chitter.rb]: I expect any `require`d modules to come in at the top

File: app/controllers/chits.rb [hunk 9]
```diff
@@ -0,0 +1,27 @@
+class Chitter < Sinatra::Base
```
Comment (sjmog) [inline app/controllers/chits.rb]: I'd expect my controllers to be classes in their own right -> we'll get to this when we look at Rails.

File: app/chitter.rb (hunk 6 above)
Comment (RBGeomaticsRob) [inline app/chitter.rb]: If it is removed the database is not setup and datamapper does not have an 'adapter' to work with?
